### PR TITLE
Add docs for the Mime component

### PR DESCRIPTION
**Pull request by fabpot**
docs for symfony/symfony#29896

### CODE REVIEW

File: components/mime.rst [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    $mimeTypes = new MimeTypes();
+    $exts = $mimeTypes->getExtensions('application/mbox');
+    // $exts contains an array of extensions: ['mbox']
+
+    $mimeTypes = $mimeTypes->getMimeTypes('mbox');
+    // $mimeTypes contains an array of mime types: ['application/mbox']
+
+    // guess a mime type for a file
+    $mimeType = $mimeTypes->guessMimeType('/some/path/to/image.gif);
```
Comment (pborreli) [inline components/mime.rst]: miss the ending quote after `gif`

File: reference/dic_tags.rst [hunk 2]
```diff
@@ -450,6 +451,15 @@ To add a new rendering strategy - in addition to the core strategies like
 :class:`Symfony\\Component\\HttpKernel\\Fragment\\FragmentRendererInterface`,
 register it as a service, then tag it with ``kernel.fragment_renderer``.
 
+mime.mime_type_guesser
+----------------------
+
+**Purpose**: Add a custom mime type guesser to MimeTypes
+
+To add a custom mime type guesser - in addition to the core guessers - create a
+class that implements :class:`Symfony\\Component\\MimeType\\MimeTypes`, register
```
Comment (Pierstoval) [inline reference/dic_tags.rst]:
```suggestion
class that implements :class:`Symfony\\Component\\Mime\\MimeTypeGuesserInterface`, register
```

**Review by Pierstoval — CHANGES_REQUESTED** [review]

File: components/mime.rst [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    namespace App;
+
+    use Symfony\Component\Mime\MimeTypeGuesserInterface;
+
+    class SomeMimeTypeGuesser implements MimeTypeGuesserInterface
+    {
+        public function isGuesserSupported(): bool
+        {
+            // return true when the guesser is supported (might depend on the OS for instance)
```
Comment (Pierstoval) [inline components/mime.rst]: What about adding this?
```suggestion
            // Return true when the guesser is supported (might depend on the OS, dependencies or PHP extensions for instance)
```

File: components/mime.rst [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            return true;
+        }
+
+        public function guessMimeType(string $path): ?string
+        {
+            // return a mime type based on the content of the file stored in $path
```
Comment (Pierstoval) [inline components/mime.rst]:
```suggestion
            // Return a mime type based on the content of the file stored in $path.
```

File: components/mime.rst [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+            // you MUST not use the filename to determine the mime type.
```
Comment (Pierstoval) [inline components/mime.rst]:
```suggestion
            // You MUST not use the filename to determine the mime type.
```

File: reference/dic_tags.rst (hunk 2 above)
Comment (pborreli) [reply reference/dic_tags.rst]: same line 24

File: components/mime.rst [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    $ composer require symfony/mime
+
+Alternatively, you can clone the `<https://github.com/symfony/mime>`_ repository.
+
+.. include:: /components/require_autoload.rst.inc
+
+Mime Types
+----------
+
+The :class:`Symfony\\Component\\MimeType\\MimeTypes` class manipulates the relationships
```
Comment (Pierstoval) [inline components/mime.rst]:
```suggestion
The :class:`Symfony\\Component\\Mime\\MimeTypes` class manipulates the relationships
```

File: components/mime.rst (hunk 1 above)
Comment (fabpot) [reply components/mime.rst]: fixed

File: reference/dic_tags.rst (hunk 2 above)
Comment (fabpot) [reply reference/dic_tags.rst]: fixed

File: components/mime.rst (hunk 5 above)
Comment (fabpot) [reply components/mime.rst]: I like it this way

File: components/mime.rst (hunk 6 above)
Comment (fabpot) [reply components/mime.rst]: fixed

**Review by OskarStark — APPROVED** [review]

File: components/mime.rst [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+Guessing the MIME type is a time-consuming process that requires inspecting
+part of the file contents. Symfony applies multiple guessing mechanisms, one
+of them based on the PHP `fileinfo extension`_. It's recommended to install
+that extension to improve the guessing performance.
+
+Adding a MIME Type Guesser
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can register your own MIME type guesser by creating a class that extends
```
Comment (tony-tran) [inline components/mime.rst]: Maybe you can change `extends` by `implements` ? WDYT ?

File: reference/dic_tags.rst [hunk 8]
```diff
@@ -450,6 +451,15 @@ To add a new rendering strategy - in addition to the core strategies like
 :class:`Symfony\\Component\\HttpKernel\\Fragment\\FragmentRendererInterface`,
 register it as a service, then tag it with ``kernel.fragment_renderer``.
 
+mime.mime_type_guesser
+----------------------
+
+**Purpose**: Add a custom MIME type guesser to MimeTypes
+
+To add a custom MIME type guesser - in addition to the core guessers - create a
+class that implements :class:`Symfony\\Component\\Mime\\MimeTypes`, register it
```
Comment (tony-tran) [inline reference/dic_tags.rst]: Is it `MimeTypesInterface` ?

File: components/mime.rst [hunk 9]
```diff
@@ -0,0 +1,88 @@
+.. index::
+    single: MIME
+    single: Components; MIME
+
+The MIME Component
+==================
+
+    The MIME component provides utilities related to MIME types, such as
+    guessing the MIME type of a given file.
```
Comment (thewilkybarkid) [inline components/mime.rst]: If the scope of the component is just media types rather than MIME as a whole, could it be named as such instead?

**Comment by OskarStark** [comment]
Looks like the code was merged, can we merge this too and announce this new component?

**Comment by javiereguiluz** [comment]
Closing because we've added all these contents to #11280, which documents the entire Mime component.

**Pull request closed without merging**